### ISSUE TITLE
Style without css loader / global css

### ISSUE THREAD

**Issue by net-bet** [fkhadra/react-toastify#525]
**Do you want to request a *feature* or report a *bug*?**
Feature
**What is the current behavior?**
If I am copying the css into the styled wrapper of the ToastContainer, the placement of the wrapper is relative to the container and not to the viewport.

Is there a possibility to allow the placement styling without the need to import the css?

**Comment by fkhadra** [comment]
Hello @net-bet ,

Can you describe your use case ?

Thanks

**Comment by net-bet** [comment]
Thanks for the response!

In order to give the basic styling I had to put *some* of the styles in the GlobalStyle (I took the css you distribute with the lib).
I don't want it in the global style to prevent possibility for future collisions.

This is my component:

  return (
    <>
      <GlobalStyle />
      <StyledToastContainer
        {...props}
        theme={theme}
      />
    </>

This is StyledToastContainer:

const StyledToastContainer = styled(ToastContainer)`
  &.Toastify__toast-container {
  }
  .Toastify__close-button {
    color: #000;
  }
  .Toastify__close-button > svg {
    width: 10px;
    height: 10px;
  }
  .Toastify__toast {
    border-radius: 6px;
    box-shadow: ${({ theme }) => `${theme.shadows.small} ${theme.palletes.dark}`};
    min-height: auto;
    max-width: 500px;
  }
  .Toastify__toast {
    &--error {
      background-color: ${({ theme }) => theme.palletes.error[5]};
    }
    &--warning{
      background-color: ${({ theme }) => theme.palletes.warning[5]};
    }
    &--success{
      background-color: ${({ theme }) => theme.palletes.success[5]};
    }
    &--info{
      background-color: ${({ theme }) => theme.palletes.info[5]};
    }
  }
  .Toastify__toast-body {}
  .Toastify__progress-bar {
    height: 3px;

    &--error{
      background: ${({ theme }) => theme.palletes.error[0]};
    }
    &--info {
      background: ${({ theme }) => theme.palletes.info[0]};
    }
    &--success {
      background: ${({ theme }) => theme.palletes.success[0]};
    }
    &--warning {
      background: ${({ theme }) => theme.palletes.warning[0]};
    }
  }
`;

This is GlobalStyle (If I move it from GlobalStyle to StyledToastContainer it doesn't work)

const GlobalStyle = createGlobalStyle`.Toastify__toast-container {
  z-index: 9999;
  -webkit-transform: translateZ(9999px);
  position: fixed;
  padding: 4px;
  width: 320px;
  box-sizing: border-box;
  color: #fff
}

.Toastify__toast-container--top-left {
  top: 1em;
  left: 1em
}

.Toastify__toast-container--top-center {
  top: 1em;
  left: 50%;
  transform: translateX(-50%)
}

.Toastify__toast-container--top-right {
  top: 1em;
  right: 1em
}

.Toastify__toast-container--bottom-left {
  bottom: 1em;
  left: 1em
}

.Toastify__toast-container--bottom-center {
  bottom: 1em;
  left: 50%;
  transform: translateX(-50%)
}

.Toastify__toast-container--bottom-right {
  bottom: 1em;
  right: 1em
}

@media only screen and (max-width:480px) {
  .Toastify__toast-container {
    width: 100vw;
    padding: 0;
    left: 0;
    margin: 0
  }
  .Toastify__toast-container--top-center,
  .Toastify__toast-container--top-left,
  .Toastify__toast-container--top-right {
    top: 0;
    transform: translateX(0)
  }
  .Toastify__toast-container--bottom-center,
  .Toastify__toast-container--bottom-left,
  .Toastify__toast-container--bottom-right {
    bottom: 0;
    transform: translateX(0)
  }
  .Toastify__toast-container--rtl {
    right: 0;
    left: auto
  }
}

.Toastify__toast {
  position: relative;
  min-height: 64px;
  box-sizing: border-box;
  margin-bottom: 1rem;
  padding: 8px;
  border-radius: 1px;
  box-shadow: 0 1px 10px 0 rgba(0, 0, 0, .1), 0 2px 15px 0 rgba(0, 0, 0, .05);
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-height: 800px;
  overflow: hidden;
  font-family: sans-serif;
  cursor: pointer;
  direction: ltr
}

.Toastify__toast--rtl {
  direction: rtl
}

.Toastify__toast--dark {
  background: #121212;
  color: #fff
}

.Toastify__toast--default {
  background: #fff;
  color: #aaa
}

.Toastify__toast--info {
  background: #3498db
}

.Toastify__toast--success {
  background: #07bc0c
}

.Toastify__toast--warning {
  background: #f1c40f
}

.Toastify__toast--error {
  background: #e74c3c
}

.Toastify__toast-body {
  margin: auto 0;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto
}

@media only screen and (max-width:480px) {
  .Toastify__toast {
    margin-bottom: 0
  }
}

.Toastify__close-button {
  color: #fff;
  background: transparent;
  outline: none;
  border: none;
  padding: 0;
  cursor: pointer;
  opacity: .7;
  transition: .3s ease;
  -ms-flex-item-align: start;
  align-self: flex-start
}

.Toastify__close-button--default {
  color: #000;
  opacity: .3
}

.Toastify__close-button>svg {
  fill: currentColor;
  height: 16px;
  width: 14px
}

.Toastify__close-button:focus,
.Toastify__close-button:hover {
  opacity: 1
}

@keyframes Toastify__trackProgress {
  0% {
    transform: scaleX(1)
  }
  to {
    transform: scaleX(0)
  }
}

.Toastify__progress-bar {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 5px;
  z-index: 9999;
  opacity: .7;
  background-color: hsla(0, 0%, 100%, .7);
  transform-origin: left
}

.Toastify__progress-bar--animated {
  animation: Toastify__trackProgress linear 1 forwards
}

.Toastify__progress-bar--controlled {
  transition: transform .2s
}

.Toastify__progress-bar--rtl {
  right: 0;
  left: auto;
  transform-origin: right
}

.Toastify__progress-bar--default {
  background: linear-gradient(90deg, #4cd964, #5ac8fa, #007aff, #34aadc, #5856d6, #ff2d55)
}

.Toastify__progress-bar--dark {
  background: #bb86fc
}

@keyframes Toastify__bounceInRight {
  0%,
  60%,
  75%,
  90%,
  to {
    animation-timing-function: cubic-bezier(.215, .61, .355, 1)
  }
  0% {
    opacity: 0;
    transform: translate3d(3000px, 0, 0)
  }
  60% {
    opacity: 1;
    transform: translate3d(-25px, 0, 0)
  }
  75% {
    transform: translate3d(10px, 0, 0)
  }
  90% {
    transform: translate3d(-5px, 0, 0)
  }
  to {
    transform: none
  }
}

@keyframes Toastify__bounceOutRight {
  20% {
    opacity: 1;
    transform: translate3d(-20px, 0, 0)
  }
  to {
    opacity: 0;
    transform: translate3d(2000px, 0, 0)
  }
}

@keyframes Toastify__bounceInLeft {
  0%,
  60%,
  75%,
  90%,
  to {
    animation-timing-function: cubic-bezier(.215, .61, .355, 1)
  }
  0% {
    opacity: 0;
    transform: translate3d(-3000px, 0, 0)
  }
  60% {
    opacity: 1;
    transform: translate3d(25px, 0, 0)
  }
  75% {
    transform: translate3d(-10px, 0, 0)
  }
  90% {
    transform: translate3d(5px, 0, 0)
  }
  to {
    transform: none
  }
}

@keyframes Toastify__bounceOutLeft {
  20% {
    opacity: 1;
    transform: translate3d(20px, 0, 0)
  }
  to {
    opacity: 0;
    transform: translate3d(-2000px, 0, 0)
  }
}

@keyframes Toastify__bounceInUp {
  0%,
  60%,
  75%,
  90%,
  to {
    animation-timing-function: cubic-bezier(.215, .61, .355, 1)
  }
  0% {
    opacity: 0;
    transform: translate3d(0, 3000px, 0)
  }
  60% {
    opacity: 1;
    transform: translate3d(0, -20px, 0)
  }
  75% {
    transform: translate3d(0, 10px, 0)
  }
  90% {
    transform: translate3d(0, -5px, 0)
  }
  to {
    transform: translateZ(0)
  }
}

@keyframes Toastify__bounceOutUp {
  20% {
    transform: translate3d(0, -10px, 0)
  }
  40%,
  45% {
    opacity: 1;
    transform: translate3d(0, 20px, 0)
  }
  to {
    opacity: 0;
    transform: translate3d(0, -2000px, 0)
  }
}

@keyframes Toastify__bounceInDown {
  0%,
  60%,
  75%,
  90%,
  to {
    animation-timing-function: cubic-bezier(.215, .61, .355, 1)
  }
  0% {
    opacity: 0;
    transform: translate3d(0, -3000px, 0)
  }
  60% {
    opacity: 1;
    transform: translate3d(0, 25px, 0)
  }
  75% {
    transform: translate3d(0, -10px, 0)
  }
  90% {
    transform: translate3d(0, 5px, 0)
  }
  to {
    transform: none
  }
}

@keyframes Toastify__bounceOutDown {
  20% {
    transform: translate3d(0, 10px, 0)
  }
  40%,
  45% {
    opacity: 1;
    transform: translate3d(0, -20px, 0)
  }
  to {
    opacity: 0;
    transform: translate3d(0, 2000px, 0)
  }
}

.Toastify__bounce-enter--bottom-left,
.Toastify__bounce-enter--top-left {
  animation-name: Toastify__bounceInLeft
}

.Toastify__bounce-enter--bottom-right,
.Toastify__bounce-enter--top-right {
  animation-name: Toastify__bounceInRight
}

.Toastify__bounce-enter--top-center {
  animation-name: Toastify__bounceInDown
}

.Toastify__bounce-enter--bottom-center {
  animation-name: Toastify__bounceInUp
}

.Toastify__bounce-exit--bottom-left,
.Toastify__bounce-exit--top-left {
  animation-name: Toastify__bounceOutLeft
}

.Toastify__bounce-exit--bottom-right,
.Toastify__bounce-exit--top-right {
  animation-name: Toastify__bounceOutRight
}

.Toastify__bounce-exit--top-center {
  animation-name: Toastify__bounceOutUp
}

.Toastify__bounce-exit--bottom-center {
  animation-name: Toastify__bounceOutDown
}

@keyframes Toastify__zoomIn {
  0% {
    opacity: 0;
    transform: scale3d(.3, .3, .3)
  }
  50% {
    opacity: 1
  }
}

@keyframes Toastify__zoomOut {
  0% {
    opacity: 1
  }
  50% {
    opacity: 0;
    transform: scale3d(.3, .3, .3)
  }
  to {
    opacity: 0
  }
}

.Toastify__zoom-enter {
  animation-name: Toastify__zoomIn
}

.Toastify__zoom-exit {
  animation-name: Toastify__zoomOut
}

@keyframes Toastify__flipIn {
  0% {
    transform: perspective(400px) rotateX(90deg);
    animation-timing-function: ease-in;
    opacity: 0
  }
  40% {
    transform: perspective(400px) rotateX(-20deg);
    animation-timing-function: ease-in
  }
  60% {
    transform: perspective(400px) rotateX(10deg);
    opacity: 1
  }
  80% {
    transform: perspective(400px) rotateX(-5deg)
  }
  to {
    transform: perspective(400px)
  }
}

@keyframes Toastify__flipOut {
  0% {
    transform: perspective(400px)
  }
  30% {
    transform: perspective(400px) rotateX(-20deg);
    opacity: 1
  }
  to {
    transform: perspective(400px) rotateX(90deg);
    opacity: 0
  }
}

.Toastify__flip-enter {
  animation-name: Toastify__flipIn
}

.Toastify__flip-exit {
  animation-name: Toastify__flipOut
}

@keyframes Toastify__slideInRight {
  0% {
    transform: translate3d(110%, 0, 0);
    visibility: visible
  }
  to {
    transform: translateZ(0)
  }
}

@keyframes Toastify__slideInLeft {
  0% {
    transform: translate3d(-110%, 0, 0);
    visibility: visible
  }
  to {
    transform: translateZ(0)
  }
}

@keyframes Toastify__slideInUp {
  0% {
    transform: translate3d(0, 110%, 0);
    visibility: visible
  }
  to {
    transform: translateZ(0)
  }
}

@keyframes Toastify__slideInDown {
  0% {
    transform: translate3d(0, -110%, 0);
    visibility: visible
  }
  to {
    transform: translateZ(0)
  }
}

@keyframes Toastify__slideOutRight {
  0% {
    transform: translateZ(0)
  }
  to {
    visibility: hidden;
    transform: translate3d(110%, 0, 0)
  }
}

@keyframes Toastify__slideOutLeft {
  0% {
    transform: translateZ(0)
  }
  to {
    visibility: hidden;
    transform: translate3d(-110%, 0, 0)
  }
}

@keyframes Toastify__slideOutDown {
  0% {
    transform: translateZ(0)
  }
  to {
    visibility: hidden;
    transform: translate3d(0, 500px, 0)
  }
}

@keyframes Toastify__slideOutUp {
  0% {
    transform: translateZ(0)
  }
  to {
    visibility: hidden;
    transform: translate3d(0, -500px, 0)
  }
}

.Toastify__slide-enter--bottom-left,
.Toastify__slide-enter--top-left {
  animation-name: Toastify__slideInLeft
}
  
.Toastify__slide-enter--bottom-right,
.Toastify__slide-enter--top-right {
  animation-name: Toastify__slideInRight
}

.Toastify__slide-enter--top-center {
  animation-name: Toastify__slideInDown
}

.Toastify__slide-enter--bottom-center {
  animation-name: Toastify__slideInUp
}

.Toastify__slide-exit--bottom-left,
.Toastify__slide-exit--top-left {
  animation-name: Toastify__slideOutLeft
}

.Toastify__slide-exit--bottom-right,
.Toastify__slide-exit--top-right {
  animation-name: Toastify__slideOutRight
}

.Toastify__slide-exit--top-center {
  animation-name: Toastify__slideOutUp
}

.Toastify__slide-exit--bottom-center {
  animation-name: Toastify__slideOutDown
}`;

**Comment by fkhadra** [comment]
@net-bet what is your setup ? CRA, nextjs ?

Could you share a codesanbox with the issue ?

**Comment by net-bet** [comment]
https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-forked-24z38?file=/src/index.js

You can see in this example that toaster toasts all the way up.
If you remove the comment from the import css everything looks good.

However, the css are identical.
I assume it is related to the place where the toaster being positioned and styled are applied should be a root level element, probably because of the keyframes or something.

Is issue clear now?

**Comment by fkhadra** [comment]
Hey @net-bet,

Thanks. I managed to fix your issue. 
[![Edit React Hook Form - Password match check - Standard Validation (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-forked-ekgp2?fontsize=14&hidenavigation=1&theme=dark)
 
Instead of using `styled(ToastContainer)` I used `createGlobalStyle` from styled-components. I believe that`styled` will inject is own classNames and it kind of create conflicts
